transactions/models.py

```diff
@@ -0,0 +1,59 @@
+from django.db import models
+
+# Create your models here.
+from blocks.models import Block
+
+
+class Transaction(models.Model):
+    '''
+    This represents a transaction(exchange of coins) on the blockchain
+    '''
+    hash = models.CharField(max_length=100, blank=True, null=True)
+    block = models.ForeignKey(Block, blank=True, null=True, on_delete=models.CASCADE, related_name='transactions')
+    tx_ins_ct = models.IntegerField(blank=True, default=0)
+    tx_outs_ct = models.IntegerField(blank=True, default=0)
+    mined = models.BooleanField(default=False)
+
+    def __str__(self):
+        return f"tx hash {self.hash} in block {self.block}"
+
+
+class AbstractTransactionComponent(models.Model):
+    '''
+    This is an abstract model specifying the common functionality
+    of transaction inputs and transaction outputs
+    '''
+    hash = models.CharField(max_length=100, blank=True, null=True)
+    value = models.IntegerField(default=0)
+    own_addr = models.CharField(max_length=100)
+    gen_tx_idx = models.IntegerField(blank=True, null=True)
```

Comment:
Same, add explanation on what they are and why they are there.

Reply:
done